edge-modules/TwinTester/ReportedPropertyOperation.cs

```diff
@@ -0,0 +1,140 @@
+    {
+        static readonly ILogger Logger = ModuleUtil.CreateLogger(nameof(ReportedPropertyOperation));
+        readonly RegistryManager registryManager;
+        readonly ModuleClient moduleClient;
+        readonly AnalyzerClient analyzerClient;
+        readonly TwinEventStorage storage;
+        TwinState twinState;
+
+        public ReportedPropertyOperation(RegistryManager registryManager, ModuleClient moduleClient, AnalyzerClient analyzerClient, TwinEventStorage storage, TwinState twinState)
+        {
+            this.registryManager = registryManager;
+            this.moduleClient = moduleClient;
+            this.analyzerClient = analyzerClient;
+            this.storage = storage;
+            this.twinState = twinState;
+        }
+
+        async Task<TwinCollection> ValidatePropertiesFromTwinAsync(Twin receivedTwin)
+        {
+            TwinCollection propertyUpdatesFromTwin = receivedTwin.Properties.Reported;
+            Dictionary<string, DateTime> reportedPropertiesUpdated = await this.storage.GetAllReportedPropertiesUpdatedAsync();
+            TwinCollection propertiesToRemoveFromTwin = new TwinCollection();
+            foreach (KeyValuePair<string, DateTime> reportedPropertyUpdate in reportedPropertiesUpdated)
+            {
+                string status;
+                if (propertyUpdatesFromTwin.Contains(reportedPropertyUpdate.Key))
+                {
+                    try
+                    {
+                        await this.storage.RemoveReportedPropertyUpdateAsync(reportedPropertyUpdate.Key);
+                    }
+                    catch (Exception e)
+                    {
+                        Logger.LogError($"Failed to remove validated reported property id {reportedPropertyUpdate.Key} from storage: {e}");
+                        continue;
+                    }
+
+                    status = $"{(int)StatusCode.Success}: Successfully validated reported property update";
+                    Logger.LogInformation(status + $" {reportedPropertyUpdate.Key}");
+                }
+                else if (TwinOperationBase.IsPastFailureThreshold(this.twinState, reportedPropertyUpdate.Value))
+                {
+                    status = $"{(int)StatusCode.ReportedPropertyUpdateNotInCloudTwin}: Failure receiving reported property update";
+                    Logger.LogError(status + $" for reported property update {reportedPropertyUpdate.Key}");
+                }
+                else
+                {
+                    continue;
+                }
+
+                propertiesToRemoveFromTwin[reportedPropertyUpdate.Key] = null; // will later be serialized as a twin update
+                await TwinOperationBase.CallAnalyzerToReportStatusAsync(this.analyzerClient, Settings.Current.ModuleId, status);
+            }
+
+            return propertiesToRemoveFromTwin;
+        }
+
+        public override async Task PerformValidationAsync()
+        {
```

Comment:
nit: when there are multiple try-catch in the same method it calls for splitting in multiple methods